### ISSUE TITLE
Decommission of legacy oemboot/netboot code

### ISSUE THREAD

**Issue by schaefi** [OSInside/kiwi#1299]
## Problem description

In KIWI we still support the pxe and oem image types to be connected to a custom KIWI initrd ```boot="netboot/..."``` or ```boot="oemboot/..."```

We kept the support because customers includes hook code which is supported by the netboot/oemboot initrd code but can't be used 1:1 when moving to dracut.

The time to phase out the support for custom KIWI initrd code has come because the distributions evolve and neither netboot nor oemboot does follow the changes anymore which will cause the code to be broken without having us noticing it.

Here is my suggestion to plan on removal of the netboot/oemboot descriptions and in the end the removal of the custom KIWI initrd build support:

- [x] Update test-image-oem-legacy to use an extra test package that provides the used oemboot description
- [x] Update test-image-pxe to be a test-image-kis or provide extra package that provides the used netboot description
- [x] Delete Leap15/15.1 netboot boot descriptions from custom_boot/ in the https://github.com/OSInside/kiwi-descriptions  repo. afaik nobody uses them
- [x] Announce the deletion of the TW oemboot descriptions from https://github.com/OSInside/kiwi-descriptions on the KIWI mailing list and Riot chat
- [x] Start a conversation with the OBS team who uses the netboot code for the buildservice workers. We would like them to take over the netboot code they use and package it for their own usecase if not already done. We would advise them to switch over to dracut and better implement a dracut module for the needs of a worker instead of using the netboot code, or switch to PXE disk image deployment for workers instead of filesystem images. That way they can plan what to do and how.
- [x] Send a drop request for kiwi-boot-descriptions in Factory. All people in suse who rely on it will now need to move. Start conversations where needed and arrange for an agreement in the same way as we did with the buildservice team
- [x] keep one version of netboot for reference, Delete custom_boot from https://github.com/OSInside/kiwi-descriptions git repo after 6months from the accept of the drop

Feedback welcome

**Comment by schaefi** [comment]
I don't have a good idea how to deal with customers that uses SLES and oemboot/netboot.

**Comment by schaefi** [comment]
I have sent an e-mail to the obs team regarding the take over of their netboot descriptions such that we can delete them from the custom_boot space but did not receive any feedback so far

**Comment by schaefi** [comment]
- [x] Update test-image-oem-legacy to use an extra test package that provides the used oemboot description
- [x] Update test-image-pxe to be a test-image-kis or provide extra package that provides the used netboot description
- [ ] Delete Leap15/15.1 netboot boot descriptions from custom_boot/ in the https://github.com/OSInside/kiwi-descriptions  repo. afaik nobody uses them
- [ ] Announce the deletion of the TW oemboot descriptions from https://github.com/OSInside/kiwi-descriptions on the KIWI mailing list and Riot chat

**Comment by schaefi** [comment]
An e-mail to the list has been sent out with the subject

```
Decommission of custom kiwi oemboot descriptions
```

oemboot is gone, thanks for the fish

**Comment by schaefi** [comment]
@davidcassany I also started a conversation with the obs team, you are CC'ed

**Comment by schaefi** [comment]
no further response from the obs team about maintaining the already pretty obs specific netboot descriptions... waiting

**Comment by schaefi** [comment]
There is also one area we haven't added to this topic. the kiwi-boot-descriptions package provides sub-packages that are used to maintain requirement lists for packages needed when building certain types of images:

kiwi-image-wsl-requires
kiwi-image-docker-requires
kiwi-image-iso-requires
kiwi-image-oem-requires
kiwi-image-pxe-requires
kiwi-image-vmx-requires
kiwi-image-tbz-requires

These sub packages are internal only and maintain requirements per image type. This concept is similar to what we might want to achieve in #1503

**Comment by schaefi** [comment]
For the September sprint this is done, I moved it respectively. But all this is not completely done. I think the next step is on refactoring of the rpm package setup as I mentioned. Once this is done there is actually no reference left except for the netboot code used by the buildservice team. As far as I understood the buildservice team already maintains their own copy of that netboot stuff to meet their individual needs. From my perspective this means we are free to drop it once we resolved the -requires helper information

So #1503 is next to touch and once this is done we will come back to this one for completion

Thanks
